authority/authority.go

```diff
@@ -80,6 +93,18 @@ func (a *Authority) init() error {
 
 	var err error
 
+	// Initialize step-ca Database if defined in configuration.
+	if a.config.DB != nil {
+		if a.db, err = a.db.Init(a.config.DB); err != nil {
+			return err
+		}
+	} else {
+		// Set the authority DB to a nil type db.DB so that we can take advantage
+		// of methods that know when the calling object is nil and respond intelligently.
+		var db *db.DB
+		a.db = db
```

Comment:
Right now only `IsRevoked` responds properly if DB is nil, it can be quite dangerous to rely on this for future changes. If you want to do something like that you probably want a noop implementation of the interface that returns whatever is expected.

Reply:
So the reason behind this was that `Revoke` should fail if DB is nil (you shouldn't be calling that method), but I can see that being confusing and not something we want to rely on.